torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -201,140 +201,231 @@ class TORCH_API ProcessGroup : public torch::CustomClassHolder {
     return size_;
   }
 
-  virtual const std::string getBackendName() const {
-    return "undefined";
-  }
+  // Subclasses must override this method to return the backend name
+  virtual const std::string getBackendName() const = 0;
 
   virtual c10::intrusive_ptr<ProcessGroup::Work> broadcast(
-      std::vector<at::Tensor>& data,
-      const BroadcastOptions& opts = BroadcastOptions()) = 0;
+      std::vector<at::Tensor>& /* tensors */,
+      const BroadcastOptions& /* opts */ = BroadcastOptions()) {
+    TORCH_CHECK(
+        false,
+        c10::str(
```

Comment:
Nit: using `fmt` would be syntactically more clean.

Reply:
Sure, will do this in the next formatting PR (will also sort the APIs).